package.json

```diff
@@ -5,6 +5,7 @@
   "scripts": {
     "build:watch": "preconstruct watch",
     "build": "preconstruct build",
+    "tsc": "tsc --build",
```

Comment:
Why do we need to build this at all? I imagine this is in order for Next.js site to "see" the packages in monorepo. I wonder if this wouldn't fix the problem: https://github.com/preconstruct/preconstruct/tree/0c794551a9158136fce27db576630d8461e44e24/packages/next

Reply:
Hmm, well I did this for typechecking reasons, not for transpilation reasons. I wanted to have `yarn tsc` in the root directory of the repository typecheck all of the code, so that you don't have to remember to run `tsc` separately in the `site` directory. Using `tsc --build` allowed me to do this since it supports `tsconfig.json` files (which may have different compiler options, as is the case here).

Reply:
I suppose it's really not necessary to have `yarn tsc` in the root directory typecheck everything. I just prefer this way because it makes it easier to verify that everything compiles before pushing to GitHub.

Reply:
For this purpose we could have a script like `"tsc": "tsc && cd site && tsc"` (or something like that). I just feel a little bit iffy about the root tsconfig having somewhat two purposes here (which manifests in the fact that `site` needs `noEmit: false` where the root tsconfig works without it).

Alternatively, we could have a root `tsconfig.json` with *just* references (and maybe some shared/base `compilerOptions`) and an additional `tsconfig.json` in `packages`). I'm not sure if it's worth the hassle though.

Reply:
Makes sense. The reason I did not go with `"tsc": "tsc && cd site && tsc"` originally is that it would prevent you from using `yarn tsc --watch`.

I just posted a third attempt at this, let me know what you think. https://github.com/emotion-js/emotion/pull/2851

Reply:
The `--watch` argument makes sense in general. I don't think we are often working on both packages and the site though in this repo. So, in this case, I prefer to have a slightly cleaner `tsconfig.json` setup.